### PR TITLE
Update REST API json specifications to 6.2

### PR DESCRIPTION
**Pull request by codebrain**

### CODE REVIEW

**Comment by codebrain** [comment]
Also includes running the API Generator tool to create the C# code.

File: src/Nest/_Generated/_Requests.generated.cs [hunk 1]
```diff
@@ -6537,8 +6563,6 @@ public partial class ValidateQueryRequest<T> : PlainRequestBase<ValidateQueryReq
 		public bool? AllowNoIndices { get => Q<bool?>("allow_no_indices"); set => Q("allow_no_indices", value); }
 		///<summary>Whether to expand wildcard expression to concrete indices that are open, closed or both.</summary>
 		public ExpandWildcards? ExpandWildcards { get => Q<ExpandWildcards?>("expand_wildcards"); set => Q("expand_wildcards", value); }
-		///<summary>TODO: ?</summary>
```
Comment (russcam) [inline src/Nest/_Generated/_Requests.generated.cs]: Needs to be patched back in, in order to not break binary compatibility

File: src/Nest/_Generated/_Requests.generated.cs [hunk 2]
```diff
@@ -6588,8 +6612,6 @@ public partial class ValidateQueryRequest : PlainRequestBase<ValidateQueryReques
...
-		public string OperationThreading { get => Q<string>("operation_threading"); set => Q("operation_threading", value); }
```
Comment (russcam) [inline src/Nest/_Generated/_Requests.generated.cs]: Needs to be patched back in, in order to not break binary compatibility

File: src/Nest/_Generated/_Requests.generated.cs [hunk 3]
```diff
@@ -5438,12 +5466,10 @@ public partial class SegmentsRequest : PlainRequestBase<SegmentsRequestParameter
...
```
Comment (russcam) [inline src/Nest/_Generated/_Requests.generated.cs]: Needs to be patched back in, in order to not break binary compatibility

File: src/Nest/_Generated/_Requests.generated.cs [hunk 4]
```diff
@@ -3825,10 +3823,8 @@ public partial class IndicesShardStoresRequest : PlainRequestBase<IndicesShardSt
...
```
Comment (russcam) [inline src/Nest/_Generated/_Requests.generated.cs]: Needs to be patched back in, in order to not break binary compatibility

File: src/Nest/_Generated/_Requests.generated.cs [hunk 5]
```diff
@@ -2701,12 +2703,8 @@ public partial class ForceMergeRequest : PlainRequestBase<ForceMergeRequestParam
 		public long? MaxNumSegments { get => Q<long?>("max_num_segments"); set => Q("max_num_segments", value); }
 		///<summary>Specify whether the operation should only expunge deleted documents</summary>
 		public bool? OnlyExpungeDeletes { get => Q<bool?>("only_expunge_deletes"); set => Q("only_expunge_deletes", value); }
-		///<summary>TODO: ?</summary>
-		public string OperationThreading { get => Q<string>("operation_threading"); set => Q("operation_threading", value); }
-		///<summary>Specify whether the request should block until the merge process is finished (default: true)</summary>
-		public bool? WaitForMerge { get => Q<bool?>("wait_for_merge"); set => Q("wait_for_merge", value); }
```
Comment (russcam) [inline src/Nest/_Generated/_Requests.generated.cs]: Needs to be patched back in, in order to not break binary compatibility

File: src/Nest/_Generated/_Requests.generated.cs [hunk 6]
```diff
@@ -2701,12 +2703,8 @@ public partial class ForceMergeRequest : PlainRequestBase<ForceMergeRequestParam
 		public long? MaxNumSegments { get => Q<long?>("max_num_segments"); set => Q("max_num_segments", value); }
 		///<summary>Specify whether the operation should only expunge deleted documents</summary>
 		public bool? OnlyExpungeDeletes { get => Q<bool?>("only_expunge_deletes"); set => Q("only_expunge_deletes", value); }
-		///<summary>TODO: ?</summary>
-		public string OperationThreading { get => Q<string>("operation_threading"); set => Q("operation_threading", value); }
```
Comment (russcam) [inline src/Nest/_Generated/_Requests.generated.cs]: Needs to be patched back in, in order to not break binary compatibility

File: src/Nest/_Generated/_Requests.generated.cs (hunk 1 above)
Comment (codebrain) [reply src/Nest/_Generated/_Requests.generated.cs]: :+1:

File: src/Nest/_Generated/_Requests.generated.cs (hunk 2 above)
Comment (codebrain) [reply src/Nest/_Generated/_Requests.generated.cs]: :+1:

File: src/Nest/_Generated/_Requests.generated.cs (hunk 3 above)
Comment (codebrain) [reply src/Nest/_Generated/_Requests.generated.cs]: :+1:

File: src/Nest/_Generated/_Requests.generated.cs (hunk 4 above)
Comment (codebrain) [reply src/Nest/_Generated/_Requests.generated.cs]: :+1:

File: src/Nest/_Generated/_Requests.generated.cs (hunk 6 above)
Comment (codebrain) [reply src/Nest/_Generated/_Requests.generated.cs]: :+1:

File: src/Nest/_Generated/_Requests.generated.cs (hunk 5 above)
Comment (codebrain) [reply src/Nest/_Generated/_Requests.generated.cs]: :+1:

**Pull request merged**